### PR TITLE
[306] Added puma ssl/https and related items

### PR DESCRIPTION
**Pull request by defong**
### Context
puma ssl/https and related items

### Changes proposed in this pull request
Added puma ssl/https and related items
default  port from 3000 to 5000
zeroed ip (0.0.0.0)
default from HTTP to HTTPS
enforced ssl usage as default

### Guidance to review

Ported mainly from https://github.com/DFE-Digital/publish-teacher-training/blob/master/config/puma.rb 

These are the minimum security and infrastructures requirements for future work.

Port 5000 was selected to remove common default rails port collusion.

### CODE REVIEW

**Comment by defong** [comment]
[SPIKE: Register integration with DfE Sign-in](https://trello.com/c/puYE8HZA/306-spike-register-integration-with-dfe-sign-in)

File: config/environments/development.rb [hunk 1]
```diff
@@ -50,4 +50,7 @@
   # Use an evented file watcher to asynchronously detect changes in source code,
   # routes, locales, etc. This feature depends on the listen gem.
   config.file_watcher = ActiveSupport::EventedFileUpdateChecker
+
+  # Force all access to the app over SSL, use Strict-Transport-Security, and use secure cookies.
+  config.force_ssl = Settings.use_ssl
```
Comment (aliuk2012) [inline config/environments/development.rb:55]: Why do we want to force ssl/https for local development?

Comment (defong) [reply config/environments/development.rb:55]: It configurable, via `Settings`.
Also being compliant with `https` is a requirement to use Dfe Signin.

**Comment by defong** [comment]
closing til, we closer

**Pull request closed without merging**